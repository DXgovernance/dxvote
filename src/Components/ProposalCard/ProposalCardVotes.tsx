import { Fragment } from 'react';
import { FiCircle } from 'react-icons/fi';
import { Loading } from 'Components/Primitives/Loading';
import {
  Detail,
  BorderedIconDetailWrapper,
  Icon,
  VoteInfoWrapper,
} from 'components/ProposalCard/ProposalCard.styled';

interface ProposalCardVotesProps {
  isLoading?: boolean;
  votes?: number[];
}

const ProposalCardVotes: React.FC<ProposalCardVotesProps> = ({
  isLoading,
  votes,
}) => {
  if (isLoading) {
    return (
      <Loading
        style={{ margin: 0 }}
        loading
        text
        skeletonProps={{ width: '200px' }}
      />
    );
  }
  return (
    <VoteInfoWrapper>
      <BorderedIconDetailWrapper>
        {votes
          .sort((a, b) => b - a)
          .map((vote, i) => {
            if (i < 3 && !(i === votes.length - 1)) {
              return (
                <Fragment>
                  <Detail>{vote}%</Detail>
                  <Icon as="div" spaceLeft spaceRight>
                    <FiCircle />
                  </Icon>
                </Fragment>
              );
            } else {
              return <Detail>{vote}%</Detail>;
            }
          })}
      </BorderedIconDetailWrapper>
    </VoteInfoWrapper>
  );
};

export default ProposalCardVotes;
