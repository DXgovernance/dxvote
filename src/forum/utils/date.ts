import moment from 'moment';

export const formatTime = date => {
  const d = new Date(date);
  return moment(d).format('MMM dd');
};

