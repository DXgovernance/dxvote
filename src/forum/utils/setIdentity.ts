import Identities from 'orbit-db-identity-provider';
import { getResolver } from '@ceramicnetwork/3id-did-resolver';

// Connects the logged in Ceramic user to OrbitDB
export async function setIdentity({ orbitdb, registry }, { ceramic, threeId }) {
  if (orbitdb.identity.type !== 'did') {
    Identities.DIDIdentityProvider.setDIDResolver(getResolver(ceramic));
    const identity = await Identities.createIdentity({
      type: 'DID',
      didProvider: threeId.getDidProvider(),
    });
    orbitdb.identity = identity;
    registry.setIdentity(identity);
    console.log('OrbitDB: DID identity set!', identity);
  }
}
