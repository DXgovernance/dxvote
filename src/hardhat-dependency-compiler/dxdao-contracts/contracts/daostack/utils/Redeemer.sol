// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/daostack/utils/Redeemer.sol';
