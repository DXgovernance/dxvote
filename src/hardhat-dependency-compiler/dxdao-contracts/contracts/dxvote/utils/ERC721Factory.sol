// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/dxvote/utils/ERC721Factory.sol';
