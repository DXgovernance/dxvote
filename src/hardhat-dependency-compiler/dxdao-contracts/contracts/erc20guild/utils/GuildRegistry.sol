// SPDX-License-Identifier: UNLICENSED
pragma solidity >0.0.0;
import 'dxdao-contracts/contracts/erc20guild/utils/GuildRegistry.sol';
