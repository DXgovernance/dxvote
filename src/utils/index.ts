export * from './address';
export * from './bignumber';
export * from './cache';
export * from './constants';
export * from './date';
export * from './enums';
export * from './helperHooks';
export * from './helpers';
export * from './number';
export * from './permissions';
export * from './proposals';
export * from './scheme';
export * from './transaction';
export * from './encodingCalls';
export * from './signature';
export * from './array';
export * from './votes';
