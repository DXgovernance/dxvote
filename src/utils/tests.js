import { ThemeProvider } from 'styled-components';
import { render as rtlRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { GuildsDarkTheme } from 'Components/theme';

export const renderWithRoutes = (ui, options = {}) => {
  //wrap with MemoryRouter
  return rtlRender(ui, { wrapper: MemoryRouter, ...options });
};

// we are mokcing testing-library/render with our own render
export const render = (children, options = {}) => {
  //wrap with Theme
  return renderWithRoutes(
    <ThemeProvider theme={GuildsDarkTheme}> {children} </ThemeProvider>,
    options
  );
};
