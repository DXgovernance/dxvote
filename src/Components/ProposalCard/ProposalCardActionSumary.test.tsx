import ProposalCardVotes from './ProposalCardVotes';
import { render } from '../../utils/tests';

jest.mock('hooks/Guilds/guild/useFilteredProposalActions', () => ({
  actions: [],
}));

test('ProposalCard votes', async () => {
  const { container } = render(
    <ProposalCardVotes isLoading={false} votes={[10, 20]} />
  );

  expect(container).toMatchSnapshot();
});

test('ProposalCard votes loading', async () => {
  const { container } = render(
    <ProposalCardVotes isLoading={true} votes={null} />
  );

  expect(container).toMatchSnapshot();
});
