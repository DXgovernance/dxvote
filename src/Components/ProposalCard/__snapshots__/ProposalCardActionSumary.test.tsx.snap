// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalCard votes 1`] = `
<div>
   
  <div
    class="sc-dlfnuX sc-fodVek jmIxJI enmJQB"
  >
    <div
      class="sc-dlfnuX sc-iqHYmW sc-bqyKOL jmIxJI eOgsPb hdYgXU"
    >
      <div
        class="sc-dlfnuX sc-crrszt jmIxJI xiXTo"
      >
        20
        %
      </div>
      <div
        class="sc-hBEYId bLtNkZ"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
        </svg>
      </div>
      <div
        class="sc-dlfnuX sc-crrszt jmIxJI xiXTo"
      >
        10
        %
      </div>
    </div>
  </div>
   
</div>
`;

exports[`ProposalCard votes loading 1`] = `
<div>
   
  <div
    class="sc-bdfBQB"
    style="margin: 0px;"
  >
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
        style="width: 200px; --base-color: #333; --highlight-color: #555;"
      >
        â€Œ
      </span>
      <br />
    </span>
  </div>
   
</div>
`;
